specs/capri/FileShare_Design.md

```diff
@@ -0,0 +1,369 @@
+# Title
+
+**Author(s)**: [Pravin Ranjan](https://github.com/PravinRanjan10)
+
+## Summary
+
+A share is a remote, mountable file system. You can mount a share to and access a share from several hosts by several users at a time. we can create a share, list shares, and show information for, update, and delete a share.To create a share, specify one of these supported protocols:
+* NFS
+* CIFS
+
+## Motivation
+
+Multiple users can access the same file share mount point.
+Data that would otherwise be duplicated on each client can be kept in a single location and accessed by client on the network.
+
+### Goals
+
+* To create and allow multiple access of a share using only one protocol(eg. NFS/CIFS)
+* To create and allow multiple access of a snapshot
+
+## Design Details
+
+1.  A user can request for action (let’s say: create a share) through
+     Northbound Plugins(NBP). We have already few plugins, but need to
+     modify for file share request.
+2. Now, NBP can communicate to OpenSDS API server.
+3. Based on user request File Share API will be selected.
+4. Internally, OpenSDS File Share API talk to controllers.
+5. Controllers will also maintain the File Share metadata in database.
+6. Controller communicates to Dock, specific driver to create a share.
+
+![FileShare Architecture Diagram](FileShare_Design.png?raw=true "File_Share_Design")
```

Comment:
Are you going to decouple file share controller from the controller as a standalone service, or just insert it like what volume controller be designed today?

Reply:
Yeah. We have target to decouple file share controller from the controller.

Reply:
Not sure. We have not decided yet.

Reply:
We should decouple.  A scheduler should be developed as well as a separate service, based on the existing Selector code.  In the future, this new scheduler can replace the Selector for block as well.